core/src/test/java/org/apache/druid/java/util/common/BytesTest.java

```diff
@@ -0,0 +1,300 @@
+import org.junit.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.groups.Default;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class BytesTest
+{
+  @Test
+  public void testNumberString()
+  {
+    Assert.assertEquals(1, Bytes.parse("1", 0L));
+    Assert.assertEquals(10000000, Bytes.parse("10000000", 0));
+  }
+
+  @Test
+  public void testWithWhiteSpace()
+  {
+    Assert.assertEquals(12345, Bytes.parse(" 12345 ", 0));
+    Assert.assertEquals(12345, Bytes.parse("\t12345\t", 0));
+  }
+
+  @Test
+  public void testK()
+  {
+    Assert.assertEquals(1000, Bytes.parse("1k", 0));
+    Assert.assertEquals(1000, Bytes.parse("1K", 0));
+  }
+
+  @Test
+  public void testM()
+  {
+    Assert.assertEquals(1000_000, Bytes.parse("1m", 0));
+    Assert.assertEquals(1000_000, Bytes.parse("1M", 0));
+  }
+
+  @Test
+  public void testG()
+  {
+    Assert.assertEquals(1000_000_000, Bytes.parse("1g", 0));
+    Assert.assertEquals(1000_000_000, Bytes.parse("1G", 0));
+  }
+
+  @Test
+  public void testT()
+  {
+    Assert.assertEquals(1000_000_000_000L, Bytes.parse("1t", 0));
+    Assert.assertEquals(1000_000_000_000L, Bytes.parse("1T", 0));
+  }
+
+  @Test
+  public void testKiB()
+  {
+    Assert.assertEquals(1024, Bytes.parse("1kib", 0));
+    Assert.assertEquals(9 * 1024, Bytes.parse("9KiB", 0));
+    Assert.assertEquals(9 * 1024, Bytes.parse("9Kib", 0));
+  }
+
+  @Test
+  public void testMb()
+  {
+    Assert.assertEquals(1024 * 1024, Bytes.parse("1mib", 0));
+    Assert.assertEquals(9 * 1024 * 1024, Bytes.parse("9MiB", 0));
+    Assert.assertEquals(9 * 1024 * 1024, Bytes.parse("9Mib", 0));
+  }
+
+  @Test
+  public void testGiB()
+  {
+    Assert.assertEquals(1024 * 1024 * 1024, Bytes.parse("1gib", 0));
+    Assert.assertEquals(1024 * 1024 * 1024, Bytes.parse("1GiB", 0));
+    Assert.assertEquals(9L * 1024 * 1024 * 1024, Bytes.parse("9Gib", 0));
+  }
+
+  @Test
+  public void testTiB()
+  {
+    Assert.assertEquals(1024 * 1024 * 1024 * 1024L, Bytes.parse("1tib", 0));
+    Assert.assertEquals(9L * 1024 * 1024 * 1024 * 1024, Bytes.parse("9TiB", 0));
+    Assert.assertEquals(9L * 1024 * 1024 * 1024 * 1024, Bytes.parse("9Tib", 0));
+  }
+
+  @Test
+  public void testPiB()
+  {
+    Assert.assertEquals(1024L * 1024 * 1024 * 1024 * 1024, Bytes.parse("1pib", 0));
+    Assert.assertEquals(9L * 1024 * 1024 * 1024 * 1024 * 1024, Bytes.parse("9PiB", 0));
+    Assert.assertEquals(9L * 1024 * 1024 * 1024 * 1024 * 1024, Bytes.parse("9Pib", 0));
+  }
+
+  @Test
+  public void testDefault()
+  {
+    Assert.assertEquals(-123, Bytes.parse(" ", -123));
+    Assert.assertEquals(-456, Bytes.parse(null, -456));
+    Assert.assertEquals(-789, Bytes.parse("\t", -789));
+  }
+
+  @Test
+  public void testException()
+  {
+    try {
+      Bytes.parse("b", 0);
+      Assert.assertFalse("IAE should be thrown", true);
```

Comment:
nit: `ExpectedException` is preferred since you can verify the error message as well as the exception type. This is useful since even when the same type of exception is thrown, exceptions from different area of code can have different messages.